ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"segmentation_id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"fixed_ips": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_address": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.SingleIP(),
+									},
+									"subnet_id": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"allowed_address_pairs": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_address": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.SingleIP(),
+									},
+									"mac_address": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"attachment_opts_vna": &schema.Schema{
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      1,
+				ConflictsWith: []string{"attachment_opts_server"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fixed_ips": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_address": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.SingleIP(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"name_other": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description_other": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags_other": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"tenant_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tenant_id_other": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"network_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"port_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceProviderConnectivityTenantConnectionV2Create(d *schema.ResourceData, meta interface{}) error {
+	var deviceType = d.Get("device_type").(string)
+	var deviceInterfaceId = d.Get("device_interface_id").(string)
+	var attachmentOpts interface{}
+
+	if deviceType != "ECL::Compute::Server" && deviceInterfaceId == "" {
```

Comment:
Why not do validation if deviceType is compute and deviceInterfaceId is not blank string?

Reply:
If the deviceType is compute, the resource will be created without any problem even if the deviceInterfaceId is not blank.